src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/DiagnosticReportMapperTest.java

```diff
@@ -1,19 +1,234 @@
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.MissingSegmentException;
+import uk.nhs.digital.nhsconnect.lab.results.utils.ResourceFullUrlGenerator;
+import uk.nhs.digital.nhsconnect.lab.results.utils.TimestampService;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
 
-import java.util.ArrayList;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.nullable;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+@SuppressWarnings("checkstyle:MagicNumber")
+@ExtendWith(MockitoExtension.class)
 class DiagnosticReportMapperTest {
 
+    @Mock
+    private UUIDGenerator uuidGenerator;
+
+    @Mock
+    private ResourceFullUrlGenerator resourceFullUrlGenerator;
+
+    @InjectMocks
+    private DiagnosticReportMapper mapper;
+
+    @Test
+    void testMapToDiagnosticReportMissingRequiredSegments() {
+        final Message message = new Message(List.of(
+            "S02+02", // ServiceReportDetails
+            "S06+06" // InvestigationSubject
+        ));
+
+        assertThatThrownBy(() -> mapper.mapToDiagnosticReport(message, null, Collections.emptyList(),
+            Collections.emptyList(), null, null, null))
+            .isExactlyInstanceOf(MissingSegmentException.class)
+            .hasMessageStartingWith("EDIFACT section is missing segment");
+    }
+
     @Test
-    void testMapMessageToDiagnosticReport() {
-        final Message message = new Message(new ArrayList<>());
+    void testMapToDiagnosticReportWithMessageWithoutResourceReferences() {
+        final Message message = new Message(List.of(
+            "S02+02",
+            "GIS+N",
+            "RFF+SRI:15/CH000042P/200010191704",
+            "STS++UN",
+            "DTM+ISR:201002251541:203"
+        ));
+
+        when(uuidGenerator.generateUUID()).thenReturn("resource-id");
+
+        final var result = mapper.mapToDiagnosticReport(message, null, Collections.emptyList(),
+            Collections.emptyList(), null, null, null);
+        ZonedDateTime dateIssued = LocalDateTime.of(2010, 02, 25, 15, 41).atZone(TimestampService.UK_ZONE);
+
+        assertAll(
+            () -> assertThat(result.getId()).isEqualTo("resource-id"),
+            () -> assertThat(result.getIdentifier())
+                .first()
+                .satisfies(identifier -> assertAll(
+                    () -> assertThat(identifier.getValue()).isEqualTo("15/CH000042P/200010191704")
+                )),
```

Comment:
Do we need assertAll here, when theres only one assertion?

Reply:
Addressed in df84be4

Apologies for not doing this yesterday